org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ScalaSourceFileEditor.scala

```diff
@@ -291,10 +290,12 @@ object ScalaSourceFileEditor {
 
   /** A thread-safe object for keeping track of Java reconciling listeners.*/
   private class ReconcilingListeners extends IJavaReconcilingListener {
-    private val reconcilingListeners = new ArrayBuffer[IJavaReconcilingListener] with SynchronizedBuffer[IJavaReconcilingListener]
+    import java.util.concurrent.ConcurrentLinkedQueue
+
+    private val reconcilingListeners = new ConcurrentLinkedQueue[IJavaReconcilingListener]()
 
     /** Return a snapshot of the currently registered `reconcilingListeners`. This is useful to avoid concurrency hazards when iterating on the `reconcilingListeners`. */
-    private def currentReconcilingListeners: List[IJavaReconcilingListener] = reconcilingListeners.toList
+    private def currentReconcilingListeners = reconcilingListeners.toArray.asInstanceOf[Array[IJavaReconcilingListener]]
```

Comment:
Maybe use `scala.collection.JavaConverters` and `asScala`? It returns `Iterable[A]` in my micro-test in console.